openml-api/src/main/java/com/feedzai/openml/provider/descriptor/fieldtype/FreeTextFieldType.java

```diff
@@ -36,18 +37,44 @@ public class FreeTextFieldType implements ModelParameterType {
      */
     private final String defaultValue;
 
+    /**
+     * A regular expression defining a valid input.
+     */
+    private final String validRegex;
+
     /**
      * Creates a new instance of this class.
      *
      * @param defaultValue  The default value.
      */
     public FreeTextFieldType(final String defaultValue) {
+        this(defaultValue, null);
+    }
+
+    /**
+     * Creates a new instance of this class.
+     *
+     * @param defaultValue  The default value.
+     * @param validRegex    A regex that matches valid inputs.
+     */
+    public FreeTextFieldType(final String defaultValue, final String validRegex) {
         this.defaultValue = Preconditions.checkNotNull(defaultValue, "defaultValue can't be null.");
+        this.validRegex = validRegex;
```

Comment:
I think we could use `Pattern.compile` to validate if the regex is valid or not, what do you think ? 

https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html#compile(java.lang.String)
It should throw a [PatternSyntaxException](https://docs.oracle.com/javase/7/docs/api/java/util/regex/PatternSyntaxException.html) if the expression's syntax is invalid.


This way we could change the `private final String validRegex` to private final Pattern validRegex

Reply:
that also prevents re-compiling the pattern multiple times

Reply:
the one potential downside is that we have to add a few more null checks

Reply:
agreed, I just implemented it